### PR TITLE
Hash an actual credential for verifyNoCredential so the timing is input-dependent

### PR DESCRIPTION
**Pull request by olorin**
@markhibberd

### CODE REVIEW

**Comment by nhibberd** [comment]
Was there any change in the benchmarks?

**Comment by olorin** [comment]
Sorry, put this on the wrong PR.

No significant change in the benchmarks.

old:

```
∃ ~/tmp/tinfoil-bench-master kdf
benchmarking kdf/scrypt/hashCredential/defaultParams
time                 1.804 s    (1.653 s .. 1.935 s)
                     0.999 R²   (0.997 R² .. 1.000 R²)
mean                 1.794 s    (1.772 s .. 1.812 s)
std dev              28.73 ms   (0.0 s .. 31.35 ms)
variance introduced by outliers: 19% (moderately inflated)

benchmarking kdf/scrypt/verifyNoCredential/defaultParams
time                 1.676 s    (1.559 s .. 1.847 s)
                     0.998 R²   (0.998 R² .. 1.000 R²)
mean                 1.872 s    (1.785 s .. 1.935 s)
std dev              94.59 ms   (0.0 s .. 108.5 ms)
variance introduced by outliers: 19% (moderately inflated)
```

new:

```
∃ ~/tmp/tinfoil-bench-no-credential kdf
benchmarking kdf/scrypt/hashCredential/defaultParams
time                 1.781 s    (1.777 s .. 1.783 s)
                     1.000 R²   (1.000 R² .. NaN R²)
mean                 1.772 s    (1.768 s .. 1.774 s)
std dev              3.777 ms   (0.0 s .. 4.261 ms)
variance introduced by outliers: 19% (moderately inflated)

benchmarking kdf/scrypt/verifyNoCredential/defaultParams
time                 1.781 s    (1.555 s .. 2.069 s)
                     0.997 R²   (0.990 R² .. 1.000 R²)
mean                 1.845 s    (1.783 s .. 1.883 s)
std dev              57.59 ms   (0.0 s .. 65.79 ms)
variance introduced by outliers: 19% (moderately inflated)
```

File: src/Tinfoil/KDF/Scrypt.hs [hunk 1]
```diff
@@ -27,10 +27,10 @@ defaultParams = scryptParams 16 8 12 -- N = 2^16, r = 8, p = 12
 salt :: IO Entropy
 salt = entropy 32
 
-verifyNoCredential :: ScryptParams -> IO (Maybe Verified)
-verifyNoCredential p = do
+verifyNoCredential :: ScryptParams -> Credential -> IO (Maybe Verified)
+verifyNoCredential p c = do
   e <- salt
```
Comment (markhibberd) [inline src/Tinfoil/KDF/Scrypt.hs]: is there a sensible way to get rid of this salt generation? I feel like verify takes long enough that we shouldn't be doing things we don't absolutely need to (i.e. isn't adding to strength of the hash). I think that using the user provided input is the right thing, but a hard coded salt of the right length doesn't affect that, we could then drop the salt from both call paths then.

Comment (olorin) [inline src/Tinfoil/KDF/Scrypt.hs]: My thoughts exactly, see #16.

**Comment by markhibberd** [comment]
👍 But I would like the salt change if there are no objections (don't mind if a different change to this though).

File: src/Tinfoil/KDF/Scrypt.hs (hunk 1 above)
Comment (markhibberd) [inline src/Tinfoil/KDF/Scrypt.hs]: Ah, yep. Sorry I missed that 💯

**Pull request merged**